frontend/lint/eslint-rules/no-unconditional-metabase-links-render.js

```diff
@@ -0,0 +1,223 @@
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+//
+// The following cases are considered errors:
+//
+// 1. MetabaseSettings.learnUrl(string)
+// 2. MetabaseSettings.docsUrl(string)
+// 3. getDocsUrl selector from "metabase/selectors/settings"
+// 4. getLearnUrl selector from "metabase/selectors/settings"
+// 5. inline string "metabase.com/docs/"
+// 6. inline string "metabase.com/learn/"
+//
+// If a link shouldn't be rendered conditionally e.g. it's only show for admins, or is rendered inside admin settings, you need to disable the rule with a reason.
+// e.g. "// eslint-disable-next-line no-unconditional-metabase-links-render -- This link only shows for admins."
+
+function getImportNodeLocation(node) {
+  return node.source.value;
+}
+
+const ADD_COMMENT_MESSAGE =
+  'add comment to indicate the reason why this rule needs to be disabled.\nExample: "// eslint-disable-next-line no-unconditional-metabase-links-render -- This links only shows for admins."';
+const ERROR_MESSAGE =
+  "Metabase links must be rendered conditionally.\n\nPlease import `getShowMetabaseLinks` selector from `metabase/selectors/whitelabel` and use it to conditionally render Metabase links.\n\nOr " +
+  ADD_COMMENT_MESSAGE;
+const LITERAL_METABASE_URL_REGEX =
+  /(metabase\.com\/docs|metabase\.com\/learn)($|\/)/;
+
+// eslint-disable-next-line import/no-commonjs
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description:
+        "Ensure that Metabase links are rendered conditionally using a `show-metabase-links` setting",
+    },
+    schema: [], // no options
+  },
+
+  create(context) {
+    let isMetabaseSettingsImported = false;
+    let isGetDocsUrlSelectorImported = false;
+    let isGetLearnUrlSelectorImported = false;
+    let isGetShowMetabaseLinksSelectorImported = false;
+
+    /**
+     * @typedef DefaultImport
+     * @property {true} isDefault
+     * @property {string} source
+     *
+     * @typedef NamedImport
+     * @property {string} named
+     * @property {string} source
+     *
+     * @param {object} node
```

Comment:
I wanted to use `ASTNode` like in the ESLint repo, but I wasn't sure how to import it. e.g. 

https://github.com/eslint/eslint/blob/57089cb5166acf8b8bdba8a8dbeb0a129f841478/lib/rules/newline-after-var.js#L75